src/calendar-app/workerUtils/worker/CalendarWorkerLocator.ts

```diff
@@ -0,0 +1,473 @@
+	locator.workerFacade = new WorkerFacade()
+	const dateProvider = new NoZoneDateProvider()
+
+	const mainInterface = worker.getMainInterface()
+
+	const suspensionHandler = new SuspensionHandler(mainInterface.infoMessageHandler, self)
+	locator.instanceMapper = new InstanceMapper()
+	locator.rsa = await createRsaImplementation(worker)
+
+	const domainConfig = new DomainConfigProvider().getCurrentDomainConfig()
+
+	locator.restClient = new RestClient(suspensionHandler, domainConfig)
+	locator.serviceExecutor = new ServiceExecutor(locator.restClient, locator.user, locator.instanceMapper, () => locator.crypto)
+	locator.entropyFacade = new EntropyFacade(locator.user, locator.serviceExecutor, random, () => locator.keyLoader)
+	locator.blobAccessToken = new BlobAccessTokenFacade(locator.serviceExecutor, dateProvider, locator.user)
+	const entityRestClient = new EntityRestClient(locator.user, locator.restClient, () => locator.crypto, locator.instanceMapper, locator.blobAccessToken)
+
+	locator.native = worker
+	locator.booking = lazyMemoized(async () => {
+		const { BookingFacade } = await import("../../../common/api/worker/facades/lazy/BookingFacade.js")
+		return new BookingFacade(locator.serviceExecutor)
+	})
+
+	let offlineStorageProvider
+	if (isOfflineStorageAvailable() && !isAdminClient()) {
+		locator.sqlCipherFacade = new SqlCipherFacadeSendDispatcher(locator.native)
+		offlineStorageProvider = async () => {
+			return new OfflineStorage(
+				locator.sqlCipherFacade,
+				new InterWindowEventFacadeSendDispatcher(worker),
+				dateProvider,
+				new OfflineStorageMigrator(OFFLINE_STORAGE_MIGRATIONS, modelInfos),
+				new CalendarOfflineCleaner(),
+			)
+		}
+	} else {
+		offlineStorageProvider = async () => null
+	}
+	locator.pdfWriter = async () => {
+		const { PdfWriter } = await import("../../../common/api/worker/pdf/PdfWriter.js")
+		return new PdfWriter(new TextEncoder(), undefined)
+	}
+
+	const maybeUninitializedStorage = new LateInitializedCacheStorageImpl(async (error: Error) => {
+		await worker.sendError(error)
+	}, offlineStorageProvider)
+
+	locator.cacheStorage = maybeUninitializedStorage
+
+	const fileApp = new NativeFileApp(new FileFacadeSendDispatcher(worker), new ExportFacadeSendDispatcher(worker))
+
+	// We don't want to cache within the admin client
+	let cache: DefaultEntityRestCache | null = null
+	if (!isAdminClient()) {
+		cache = new DefaultEntityRestCache(entityRestClient, maybeUninitializedStorage)
+	}
+
+	locator.cache = cache ?? entityRestClient
+
+	locator.cachingEntityClient = new EntityClient(locator.cache)
+	const nonCachingEntityClient = new EntityClient(entityRestClient)
+
+	locator.cacheManagement = lazyMemoized(async () => {
+		const { CacheManagementFacade } = await import("../../../common/api/worker/facades/lazy/CacheManagementFacade.js")
+		return new CacheManagementFacade(locator.user, locator.cachingEntityClient, assertNotNull(cache))
+	})
+
+	if (isIOSApp() || isAndroidApp()) {
```

Comment:
It should always be an iOS or Android app as well, right?

Reply:
I believe there is still a possibility of making a standalone desktop/web app so I would like to leave this stuff in.